packages/execution-environments/src/common/BaseSnapExecutor.test.ts

```diff
@@ -708,6 +727,47 @@ describe('BaseSnapExecutor', () => {
     });
   });
 
+  it('supports getTransactionInsight export', async () => {
+    const CODE = `
+      module.exports.getTransactionInsight = ({ transaction }) => transaction;
+    `;
+    const executor = new TestSnapExecutor();
+
+    await executor.writeCommand({
+      jsonrpc: '2.0',
+      id: 1,
+      method: 'executeSnap',
+      params: [FAKE_SNAP_NAME, CODE, []],
+    });
+
+    expect(await executor.readCommand()).toStrictEqual({
+      jsonrpc: '2.0',
+      id: 1,
+      result: 'OK',
+    });
+
+    // @todo
```

Comment:
what's the todo here?

Reply:
Updating the value to be closer to the value we expect from the extension. But we have not decided on the shape of that object yet I believe.